src/Orleans/GrainDirectory/MultiClusterStatus.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.GrainDirectory
+{
+    /// <summary>
+    /// Status of a directory entry with respect to multi-cluster registration
+    /// </summary>
+    internal enum MultiClusterStatus
```

Comment:
The MultiClusterStatus enum should probably inherit from byte, since that is also how you are serializing it in the protocol